openedx_tagging/core/tagging/rest_api/v1/views_import.py

```diff
@@ -49,3 +53,51 @@ def get(self, request: Request, file_ext: str, *args, **kwargs) -> FileResponse:
         response = FileResponse(fh, content_type=content_type)
         response['Content-Disposition'] = content_disposition
         return response
+
+
+class ImportView(APIView):
+    """
+    View to import taxonomies
+
+    **Example Requests**
+        POST /tagging/rest_api/v1/import/
+        {
+            "taxonomy_name": "Taxonomy Name",
+            "taxonomy_description": "This is a description",
+            "file": <file>,
+        }
+
+    **Query Returns**
+        * 200 - Success
+        * 400 - Bad request
+        * 405 - Method not allowed
+    """
+    http_method_names = ['post']
```

Comment:
We're going to need to the REST views to support both:

* POST - create new taxonomy + import tags
* PUT - update existing taxonomy + import tags

So I thought that the `POST` view could call both `create_taxonomy` and `import_tags` (and `delete_taxonomy` if it failed), and the `PUT` view could just call `import_tags`.

And then we don't need the `create_taxonomy_and_import_tags` API method here or on the platform side.

Reply:
Made a refactor here: a50e08f1cf1efd3bc92078a6d29a8decc91bc929
I moved the View class to be actions inside our TaxonomyView. That way we can inherit some permissions [here](https://github.com/openedx/openedx-learning/pull/112/files#r1384924676)